include/i2c.h

```diff
@@ -415,17 +477,8 @@ static inline int i2c_burst_read(struct device *dev, u16_t dev_addr,
 				 u8_t start_addr, u8_t *buf,
```

Comment:
Functions should be deprecated using `__deprecated` pragma.

Reply:
OK.  It'll be nice when the deprecation policy and process gets documented somewhere.